common/src/main/java/io/homeassistant/companion/android/common/data/websocket/impl/WebSocketRepositoryImpl.kt

```diff
@@ -128,21 +173,36 @@ class WebSocketRepositoryImpl @Inject constructor(
     }
 
     @ExperimentalCoroutinesApi
-    override suspend fun getStateChanges(): Flow<StateChangedEvent> {
-        stateChangedMutex.withLock {
-            if (stateChangedFlow == null) {
+    override suspend fun getStateChanges(): Flow<StateChangedEvent> = subscribeToEventsForType("state_changed")
```

Comment:
I'd have made all the type strings that you subscribe to into constants - that way there is no chance of a typo when you come to handle them in handleEvent().

Reply:
There aren't any typos right now :) But it is nice to have when used in more than one place and it looks like the iOS websocket implementation also uses constants for this situation - changed.